node/localnode.go

```diff
@@ -207,6 +212,14 @@ func (localNode *LocalNode) maybeAddRemoteNode(remoteNode *nnetnode.RemoteNode)
 	return nil
 }
 
+func (localNode *LocalNode) GetProposalSubmitted() uint32 {
+	return localNode.proposalSubmitted
```

Comment:
GetProposalSubmitted() and IncrementProposalSubmitted() can be called together, so a lock is needed.

Reply:
I consider solidly consistency is unnecessary at here.
localNode.proposalSubmitted just only has one writer in a single goroutine consensus.startProposing(), and never should be in concurrent writing. (concurrent proposal is nonlicet and will be rejected by neighbors)
On the other hand, proposalSubmitted just a reference for evaluate node's propose success rate. Deviation 1 with probability is acceptable. Whatever, getnodeinfo RPC should not influenced the performance of consensus routine for avoid potential DDOS attack.

Reply:
Anyway, I push a commit for make sure atomic R/W localNode.proposalSubmitted without lock.